examples/webhook-authz/README.md

```diff
@@ -0,0 +1,106 @@
+# Example Webhook authorizer
+
+This directory contains the scaffolding necessary to use a webhook authorizer
+with the standalone cluster registry. It is not meant to be used with the
+aggregated cluster registry.
+
+NOTE: this example is not meant for production use! Do not deploy this into a
+production environment.
+
+## Usage
+
+1.  Deploy a standalone cluster registry. You can use
+    [crinit](https://github.com/kubernetes/cluster-registry/blob/master/docs/userguide.md#standalone).
+
+2.  Create a self-signed SSL cert for the cluster registry webhook authorizer
+    (or skip this step and use an existing SSL cert).
+
+```sh
+openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -nodes
+```
+
+3.  Create a ConfigMap that contains the files needed for the cluster registry
+    webhook authorizer. If you deployed your cluster registry into a namespace
+    other than `clusterregistry`, change the namespace below accordingly.
+
+```sh
+kubectl create configmap bitesize -n clusterregistry \
+  --from-file=./rules.hcl --from-file=./webhook-config.yaml \
+  --from-file=./key.pem --from-file=./cert.pem
+```
+
+4.  Create the webhook authorizer. Note that you may need to change the
+    namespace if you did not deploy the cluster registry into the
+    `clusterregistry` namespace.
+
+```sh
+kubectl apply -f ./authz.yaml -n clusterregistry
+```
+
+5.  Update the cluster registry deployment as follows.
+
+```yaml
+...
+      - command:
+        - ./clusterregistry
+        - --advertise-address=<...>
+        - --bind-address=0.0.0.0
+        - --client-ca-file=/etc/clusterregistry/apiserver/ca.crt
+        - --etcd-servers=http://localhost:2379
+        - --secure-port=8443
+        - --tls-cert-file=/etc/clusterregistry/apiserver/server.crt
+        - --tls-private-key-file=/etc/clusterregistry/apiserver/server.key
+        - --token-auth-file=/etc/clusterregistry/apiserver/token.csv
+        - --use-delegated-auth=false
+        - --authorization-always-allow=false
+        - --authorization-webhook=true
+        - --authorization-webhook-config-file=/etc/bitesize/webhook-config.yaml
+        - --authorization-webhook-cache-authorized-ttl=5m
+        - --authorization-webhook-cache-unauthorized-ttl=30s
+        volumeMounts:
+          - name: bitesize
+            mountPath: /etc/bitesize
+          - name: ca-certificates
+            mountPath: /etc/ssl/certs
+...
+      volumes:
+      - name: bitesize
+        configMap:
+          name: bitesize
+      - name: ca-certificates
+        hostPath:
+          path: /etc/ssl/certs
+...
+```
+
+If everything is running properly, you should be able to access clusters as the
+admin user:
+
+```sh
+$ kubectl get clusters --context <your_cluster_registry_context>
+No resources found
```

Comment:
What infrastructure are you testing this on? Using the k8s `local-up-cluster.sh` I am getting:

```
Error from server (Forbidden): clusters.clusterregistry.k8s.io is forbidden: User "system:anonymous" cannot list clusters.clusterregistry.k8s.io at the cluster scope: Not allowed
```

Reply:
We've been debugging this offline. It seems like something odd is happening, but it's not quite clear what. It might have to do with the standalone vs aggregated authn/authz handling.